Tests/LibAudio/TestWav.cpp

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2024, Lee Hanken
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/ByteBuffer.h>
+#include <AK/LexicalPath.h>
+#include <LibAudio/WavLoader.h>
+#include <LibAudio/WavWriter.h>
+#include <LibCore/Directory.h>
+#include <LibCore/System.h>
+#include <LibTest/TestCase.h>
+
+static void run_test(StringView const file_name, int const num_samples, int channels, u32 const rate, u32 const checksum)
+{
+
+    const ByteString format = "RIFF WAVE (.wav)";
+    const int bits = 16;
+
+    // This makes sure that the tests will run both on target and in Lagom.
+    #ifdef AK_OS_SERENITY
+        ByteString path = ByteString::formatted("/usr/Tests/LibAudio/wav-test-files/{}", file_name);
+        ByteString out_path = ByteString::formatted("/home/anon/saved_{}", file_name);
+    #else
+        ByteString path = ByteString::formatted("wav-test-files/{}", file_name);
+        ByteString out_path = ByteString::formatted("saved_{}", file_name);
+    #endif
+
+    auto maybe_loader = Audio::Loader::create(path);
+    if (maybe_loader.is_error()) {
+        FAIL(maybe_loader.error());
+        return;
+    }
+    auto loader = maybe_loader.value();
+
+    EXPECT_EQ(loader->format_name(), format);
+    EXPECT_EQ(loader->sample_rate(), rate);
+    EXPECT_EQ(loader->num_channels(), channels);
+    EXPECT_EQ(loader->bits_per_sample(), bits);
+    EXPECT_EQ(loader->total_samples(), num_samples);
+
+    auto maybe_writer = Audio::WavWriter::create_from_file(out_path, rate, channels);
+    if (maybe_writer.is_error()) {
+        FAIL(maybe_writer.error());
+        return;
+    }
+    auto writer = maybe_writer.release_value();
+
+    u32 sum=0;
+    int samples_read=0;
+    int size=0;
+    do {
+        auto maybe_samples = loader->get_more_samples(1000);
```

Comment:
Not sure what the custom sample count achieves here. For test code it's usually best to stick with the defaults.